qiskit/pulse/builder.py

```diff
@@ -351,10 +371,21 @@ def backend(self):
         """
         return self._backend
 
-    @property
-    def context_schedule(self) -> Schedule:
-        """Return the current context schedule."""
-        return self._context_schedule
+    @_compile_lazy_circuit_before
+    def push_context(self, alignment: AlignmentKind):
+        """Push new context to the stack."""
+        self._context_stack.append(ScheduleBlock(alignment_context=alignment))
+
+    @_compile_lazy_circuit_before
+    def pop_context(self):
+        """Pop the last context from the stack and append it to the parent."""
```

Comment:
The "and append to parent" seems uncommon for `pop`. Is there another term we can use for this?

This made `compile` below very difficult to understand:
```
        while len(self._context_stack) > 1:
            self.pop_context()
```
without reading the implementation of `pop_context`, it looks like this would just remove everything.

Reply:
Umm you're right. Any suggestion? `pop_append_context`?

Reply:
Yes, that's definitely better. 🤔 Not sure I have a better suggestion. `process_last`? @eggerdj, ideas?

Reply:
I lean toward updating the behavior of method. `push ` & `pop` are standard terms for stack data structure and stick to these names would be better for developers. Instead of internally appending to parent, we can call `builder.append_block` method externally.